internal/app/osd/internal/reg/reg.go

```diff
@@ -397,3 +400,66 @@ func (r *Registrator) Top(ctx context.Context, in *empty.Empty) (reply *proto.To
 	reply = &proto.TopReply{ProcessList: p}
 	return
 }
+
+// DF implements the proto.OSDServer interface.
+func (r *Registrator) DF(ctx context.Context, in *empty.Empty) (reply *proto.DFReply, err error) {
+	var stat unix.Statfs_t
+	file, err := os.Open("/proc/mounts")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer file.Close()
+
+	stats := []*proto.DFStat{}
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		fields := strings.Fields(scanner.Text())
+
+		filesystem := fields[0]
+		mountpoint := fields[1]
+
+		f, err := os.Stat(mountpoint)
+		if err != nil {
+			log.Fatal(err)
```

Comment:
log.Fatal terminates the process, this is not what you probably want to do? (there're several of them in this function)

https://godoc.org/log#Fatal

Reply:
Ahh yep. I was developing this as an independent binary for testing purposes. Forgot to remove.

Reply:
isn't it better to return error back to the caller?

or in this particular case, to skip mount point, or extend response to include error per mountpoint, so that we can report that mountpoint does exist, but getting `df` for it results in error?

or even use multierror and return (partial) response and error (I hope gRPC can do that)?

Reply:
not sure gRPC has an option for partial response 🤔. We can probably just skip the mount for now.

Reply:
I mean if you return `resp, err` back to the caller, and if gRPC transports it correctly back, you can show back response (if non-nil) and error independently, like real `df` does afaik. It shows output and any errors encountered

Reply:
I see. Then yes that _should_ work. It is up to the caller to check the return value but I'm not sure I can get gRPC to return multi-error  type.

Reply:
probably gRPC would cast this to string and back, but this should probably still good enough